### PR TITLE
Restore headroom for additive attention masks

### PR DESCRIPTION
**Pull request by Acentelles**
## Topic and evidence summary

**Review group:** Correctness and verifier portability. **Type:** Bug fix.

Restore additive-mask headroom: a masked score must not win merely because the sentinel was derived only from the exponential cutoff. The finite-mask condition and integer range still apply.

**Baseline and observed effect.** No performance optimization claimed. Correctness baseline: scores [0,20] plus the old -12 mask become [0,8], allowing the masked position to win; the regression checks zero probability at masked positions with restored headroom. Quantized models and registrations must be regenerated.

**Dependencies.** This PR targets #288 (`perf/round-2`); reviewed head `286e433558db`, code base `29170bd3a813`. The benchmark baseline is the configuration described above and below, which may include other integrated changes. The topic group does not change merge order. Related PRs: #293, #294, #335.

## Problem and change

The small attention-mask sentinel is chosen only from the exponential table's cutoff. That does not make it a valid additive mask for arbitrary attention scores. At scale 14, scores `[0, 20]` with the second position masked become `[0, 8]` after adding the current `-12` sentinel. The masked position wins.

Restore the earlier `-2^30` integer headroom for extreme negative mask constants. Correct the documentation to state the actual finite-mask condition: its magnitude must exceed the score difference plus the exponential cutoff, and the addition must stay in range. This is still a finite approximation to negative infinity, not an unconditional guarantee for all representable scores.

Add a regression with a larger masked score and an Atlas proof regression that adds the mask before softmax. Update the older attention fixture to use the restored sentinel. The current generic significance clamp already handles the larger gap.

## Validation

- A standalone native proof harness constructs two attention rows with one unmasked score of zero and seven masked scores of twenty. The resulting mask/add/softmax graph assigns zero probability to every masked position.
- Its complete ordinary HyperKZG proof verifies, and a changed output is rejected. The proof serializes to 26,445 bytes. This uses Atlas test setup for a correctness regression.
- A full Qwen2-0.5B forward pass completes with the change. It still disagrees with the floating-point reference on the recorded prompt's next token, so this PR does not claim to resolve Qwen's broader quantization error.

The native proof build uses pinned Tract and the existing arkworks compatibility export. Cached quantized models and their registrations need regeneration because the affected constant values change.

### CODE REVIEW

File: atlas-onnx-tracer/src/utils/quantize.rs [hunk 1]
```diff
@@ -78,18 +78,18 @@ pub fn quantize_tensor(tensor: Tensor<f32>, scale: Scale) -> Tensor<i32> {
...
+/// Float magnitude used when an extreme negative mask cannot be represented.
+///
+/// Keep half the signed integer range available for an additive mask. A mask
+/// must dominate the difference between masked and unmasked attention scores,
+/// not just the exponential table's cutoff. A small sentinel derived only from
+/// that cutoff can let a masked score become the row maximum.
+///
+/// This remains a finite approximation to negative infinity. Correct masking
+/// requires the sentinel to exceed the score difference plus the exp cutoff;
+/// callers must also keep the score-plus-mask addition within the integer range.
```
[the remark this answers is not in the document]
Comment (AntoineF4C5) [reply atlas-onnx-tracer/src/utils/quantize.rs:90]: Addressed in #366.

**Comment by Acentelles** [comment]
Consolidated into https://github.com/ICME-Lab/jolt-atlas/pull/369.

The replacement preserves this change and its recorded evidence. Closing this PR retires the duplicate review queue. The original branch and reviewed head `5f0024556c04cb3d1880273f90aede18691f7faa` remain available.

[Review order, source map and validation](https://github.com/ICME-Lab/jolt-atlas/blob/review/consolidation-records/docs/review-2026-09-21/README.md). We are keeping five replacement PRs open; the native prover and verifier branches will become PRs as earlier reviews merge.

The original base branch remains available for #366. Its centering commit is included with authorship preserved; #366 itself is unchanged.
